examples/bq-log-alerting/example-foundation/main.tf

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  alias = "impersonate"
```

Comment:
This is a best practice, but we tend to keep examples tightly scoped and simple. I would recommend we use the normal CFT practice for examples like https://github.com/terraform-google-modules/terraform-google-project-factory/tree/master/examples/simple_project

Reply:
The example in the link is simple and straightforward, but it uses a JSON service account credentials file to set the provider credentials:

```
provider "google" {
  credentials = file(local.credentials_file_path)
  version     = "~> 3.30"
}
```
and we want to be able to deploy this solution in the `terraform-example-foundation` without much extra effort from the user.
The org policies set in the foundation do not allow the creation of a service account credential file, which would force the user to use this "impersonate" practice to be able to deploy the solution example.

I think it would help the user if we keep this version.

Reply:
>  it uses a JSON service account credentials file

We don't need this either. I just meant if we introduce the impersonation it may be unnecessarily complex for just showcasing this module. Perhaps something like [this](https://github.com/terraform-google-modules/terraform-google-project-factory/blob/master/examples/shared_vpc/main.tf)?

> we want to be able to deploy this solution in the terraform-example-foundation without much extra effort from the user

My thought process was this is a standalone example and has no relation to example foundations. In example foundations, we will directly invoke and use the module and not use this example. The purpose of this example would be just a simple demo of the module. We can sync tomorrow if you had a different approach

Reply:
Fixed